### ISSUE TITLE
CNN - convolution layer 필터(kernel) 적용

### ISSUE THREAD

**Issue by SeBin7** [SeBin7/Road_Vision#22]
1. 

<img width="937" height="543" alt="Image" src="https://github.com/user-attachments/assets/d3438558-8541-41a8-a740-5e307bba2d21" />

<img width="937" height="543" alt="Image" src="https://github.com/user-attachments/assets/89d75dea-a98c-4a70-afcf-4087de039de9" />

[Epoch 1] 평균 Loss: 0.6331, Acc: 0.7003

[Epoch 2] 평균 Loss: 0.4847, Acc: 0.7669

주로 자동차 객체와 라인 위주로  특징 추출 

```
#CNN.py
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.nn.init as init

class CNNFeatureExtractor(nn.Module):
    def __init__(self, feature_dim=128):
        super().__init__()

        # Conv Layer 정의
        self.conv_layers = nn.Sequential(
            nn.Conv2d(3, 16, 3, padding=1),  # conv0: 엣지 필터로 초기화
            nn.ReLU(),
            nn.MaxPool2d(2),

            nn.Conv2d(16, 32, 3, padding=1),
            nn.ReLU(),
            nn.MaxPool2d(2),

            nn.Conv2d(32, 64, 3, padding=1),
            nn.ReLU(),
            nn.MaxPool2d(2)
        )

        self.fc = nn.Linear(64 * 28 * 28, feature_dim)

        # ✅ 첫 번째 conv 필터를 Sobel edge filter로 초기화
        self._init_sobel_filter()

    def _init_sobel_filter(self):
        """Conv1의 weight를 Sobel 필터 기반으로 초기화"""
        sobel_x = torch.tensor([[-1., 0., 1.],
                                [-2., 0., 2.],
                                [-1., 0., 1.]])
        sobel_y = torch.tensor([[-1., -2., -1.],
                                [ 0.,  0.,  0.],
                                [ 1.,  2.,  1.]])

        conv0 = self.conv_layers[0]  # Conv2d(3, 16, 3, padding=1)

        # 필터 weight 초기화: shape = [out_channels, in_channels, kernel_h, kernel_w]
        with torch.no_grad():
            for i in range(16):  # 16개 필터 각각 초기화
                if i % 2 == 0:
                    kernel = sobel_x.clone()
                else:
                    kernel = sobel_y.clone()

                for c in range(3):  # RGB 각각 동일 kernel 적용
                    conv0.weight[i, c] = kernel

            conv0.bias.zero_()  # bias 0으로 초기화

    def forward_features(self, x):
        return self.conv_layers(x)

    def forward(self, x, flatten=True):
        x = self.conv_layers(x)
        if flatten:
            x = x.view(x.size(0), -1)
            x = self.fc(x)
        return x
```
